Slim/Plugin/Podcast/Plugin.pm

```diff
@@ -183,23 +185,39 @@ sub handleFeed {
 			playlist => $url,
 		};
 
+		# if pre-cached feed data is missing, initiate retrieval
 		unless ($image && $cache->get('podcast_moreInfo_' . $url)) {
-			Slim::Formats::XML->getFeedAsync(
-				sub {
-					_precacheShowDetails($url, $_[0]);
-				},
-				sub {
-					$log->warn("can't get $url RSS feed information: ", $_[0]);
-				},
-				{
-					parser => 'Slim::Plugin::Podcast::Parser',
-					url => $_->{value},
-					expires => 86400
-				}
-			);
+			# cache a placeholder image & moreInfo to guard against retrieving
+			# the feed multiple times while browsing within the podcast menu
+			# they will be replaced with real data after feed is successfully retrieved
+			$cache->set('podcast-rss-' . $url, __PACKAGE__->_pluginDataFor('icon'), '1days');
+			$cache->set('podcast_moreInfo_' . $url, {}, '1days');
+			push (@need, $url);
 		}
 	}
 
+	# get missing cache images & moreinfo if any
+	# each feed is retrieved and parsed sequentially, to limit loading on modestly powered servers
+	$fetch = sub {
+		my $url = pop @need;
+		Slim::Formats::XML->getFeedAsync(
+			sub {
```

Comment:
In that case we don't even need to create a sub here, we can just use $fetch

Reply:
I'm not seeing your point.

Reply:
I mean
```
	$fetch = sub {
		my $url = pop @need;
		Slim::Formats::XML->getFeedAsync(
			$fetch,
			sub {
				$log->warn("can't get $url RSS feed information: ", $_[0]);
				$fetch->();
			},
			{
				parser => 'Slim::Plugin::Podcast::Parser',
				url => $url,
				expires => 86400,
			}
		) if $url;
	};
```

Reply:
Yes, I see now. One line can replace five lines.
Is there any other advantage ?

For some reason the intent doesn't seem quite as clear to me, so I'd leave as is, unless there is another advantage. But I don't feel strongly about it.

Reply:
I don't mind the few lines. Plus it protects the call from changes where we would start to accept some input parameter in `$fetch`.